hedera-mirror-rest-java/src/main/java/com/hedera/mirror/restjava/common/LinkFactoryImpl.java

```diff
@@ -62,39 +64,79 @@ private <T> String createNextLink(
+        builder.queryParams(linkedMultiValueMap);
+        return builder.toUriString();
+    }
+
+    private void addParamMapToLink(
+            Map<String, String[]> paramsMap,
+            Map<String, String> paginationParamsMap,
+            Direction order,
+            LinkedMultiValueMap<String, String> linkedMultiValueMap) {
         for (var entry : paramsMap.entrySet()) {
             var key = entry.getKey();
             if (!paginationParamsMap.containsKey(key)) {
-                builder.queryParam(key, (Object[]) entry.getValue());
+                for (var value : entry.getValue()) {
+                    linkedMultiValueMap.add(entry.getKey(), value);
+                }
             } else {
-                addQueryParamToLink(entry, builder, order);
+                addQueryParamToLink(entry, order, linkedMultiValueMap);
             }
         }
-
-        for (var entry : paginationParamsMap.entrySet()) {
-            var key = entry.getKey();
-            var exclusive = exclusiveParam == null || key.equals(exclusiveParam);
-            builder.queryParam(key, getOperator(order, exclusive) + ":" + entry.getValue());
-        }
-
-        return builder.toUriString();
     }
 
-    private void addQueryParamToLink(Entry<String, String[]> entry, UriComponentsBuilder builder, Direction order) {
+    private void addQueryParamToLink(
+            Entry<String, String[]> entry, Direction order, LinkedMultiValueMap<String, String> linkedMultiValueMap) {
         for (var value : entry.getValue()) {
             // Skip if it's in the same direction as the order, the new bound should come from the extracted value
             if (isSameDirection(order, value)) {
                 continue;
             }
 
-            builder.queryParam(entry.getKey(), value);
+            linkedMultiValueMap.add(entry.getKey(), value);
+            if (hasEq(value)) {
+                // Only 1 eq is allowed, return without adding additional values
+                return;
+            }
+        }
+    }
+
+    private void addExtractedParamsToLink(
+            Sort sort,
+            Map<String, String> paginationParamsMap,
+            Direction order,
+            LinkedMultiValueMap<String, String> linkedMultiValueMap) {
+        var sortKeys = sort.map(Sort.Order::getProperty).toList();
+        if (sortKeys.isEmpty()) {
+            for (var entry : paginationParamsMap.entrySet()) {
+                linkedMultiValueMap.add(entry.getKey(), entry.getValue());
+            }
+            return;
+        }
+
+        for (int i = 0; i < sortKeys.size(); i++) {
+            var key = sortKeys.get(i);
+            if (linkedMultiValueMap.containsKey(key) && containsEq(linkedMultiValueMap.get(key))) {
+                // This query parameter has already been added with an eq
+                continue;
+            }
+
+            var entry = paginationParamsMap.get(key);
```

Comment:
```suggestion
            var value = paginationParamsMap.get(key);
```

since `entry` is misleading.

Reply:
Thanks I have updated it.